internal/controller/bootstrap/init.go

```diff
@@ -254,10 +255,22 @@ func (b *Bootstrap) InitResources(instance *apiv3.CommonService, forceUpdateODLM
 		}
 	}
 
-	klog.Info("Installing ODLM Operator")
-	if err := b.renderTemplate(constant.ODLMSubscription, b.CSData); err != nil {
+	// Install ODLM Operator
+	// Check if CatalogSource contains the correct version of ODLM
+	// if contains, install ODLM Operator
+	// if not, skip the installation of ODLM Operator, and show warning event
+	installODLM, err := util.CheckODLMCatalogSource(b.Reader, b.CSData.ODLMCatalogSourceName, b.CSData.ODLMCatalogSourceNs)
+	if err != nil {
 		return err
 	}
+	if installODLM {
+		klog.Info("Installing ODLM Operator")
+		if err := b.renderTemplate(constant.ODLMSubscription, b.CSData); err != nil {
+			return err
+		}
+	} else {
+		b.EventRecorder.Event(instance, corev1.EventTypeWarning, "ODLMCatalogSourceWarning", fmt.Sprintf("The catalogsource %s in namespace %s does not contain the correct version of ODLM, skip the installation/update of ODLM Operator", b.CSData.ODLMCatalogSourceName, b.CSData.ODLMCatalogSourceNs))
```

Comment:
Do we need to return error if the catalog does not contain the expected ODLM channel?